packages/ts-moose-lib/src/clients/redisClient.ts

```diff
@@ -0,0 +1,236 @@
+import { createClient, RedisClientType } from "redis";
+
+export class MooseCache {
+  private static instance: MooseCache;
+  private client: RedisClientType;
+  private isConnected: boolean = false;
+  private readonly keyPrefix: string;
+  private disconnectTimer: NodeJS.Timeout | null = null;
+  private readonly idleTimeout: number;
+
+  private constructor() {
+    const redisUrl =
+      process.env.MOOSE_REDIS_CONFIG__URL || "redis://127.0.0.1:6379";
+    const prefix = process.env.MOOSE_REDIS_CONFIG__KEY_PREFIX || "MS";
+    // 30 seconds of inactivity before disconnecting
+    this.idleTimeout =
+      parseInt(process.env.MOOSE_REDIS_CONFIG__IDLE_TIMEOUT || "30", 10) * 1000;
+    this.keyPrefix = `${prefix}::moosecache::`;
+
+    this.client = createClient({
+      url: redisUrl,
+    });
+
+    this.client.on("error", (err: Error) => {
+      console.error("TS Redis client error:", err);
+      this.isConnected = false;
+      this.clearDisconnectTimer();
+    });
+
+    this.client.on("connect", () => {
+      this.isConnected = true;
+      console.log("TS Redis client connected");
+    });
+
+    this.client.on("end", () => {
+      this.isConnected = false;
+      console.log("TS Redis client disconnected");
+      this.clearDisconnectTimer();
+    });
+  }
+
+  private clearDisconnectTimer(): void {
+    if (this.disconnectTimer) {
+      clearTimeout(this.disconnectTimer);
+      this.disconnectTimer = null;
+    }
+  }
+
+  private resetDisconnectTimer(): void {
+    this.clearDisconnectTimer();
+    this.disconnectTimer = setTimeout(async () => {
+      if (this.isConnected) {
+        console.log("TS Redis client disconnecting due to inactivity");
+        await this.disconnect();
+      }
+    }, this.idleTimeout);
+  }
+
+  private async ensureConnected(): Promise<void> {
+    if (!this.isConnected) {
+      await this.connect();
+    }
+    this.resetDisconnectTimer();
+  }
+
+  private async connect(): Promise<void> {
+    if (!this.isConnected) {
+      await this.client.connect();
+      this.resetDisconnectTimer();
+    }
+  }
+
+  private getPrefixedKey(key: string): string {
+    return `${this.keyPrefix}${key}`;
+  }
+
+  /**
+   * Gets the singleton instance of MooseCache. Creates a new instance if one doesn't exist.
+   * The client will automatically connect to Redis and handle reconnection if needed.
+   *
+   * @returns Promise<MooseCache> The singleton instance of MooseCache
+   * @example
+   * const cache = await MooseCache.get();
+   */
+  public static async get(): Promise<MooseCache> {
```

Comment:
Did you double check that this was indeed making a singleton? I would do it differently

ie have an variable top of the file and return its value